File: Week 1/PSet/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int length(long);
long power(long,long);
long breakdown(long);

int main(void)
{
    long number = get_long("Number: ");

    int n = length(number);

    long a;
    long b;

    long sum = 0;

    for (int i = 1; i < n+1; ++i)
    {
        long x = number % power(10.0,(long) i);

        x = x / power(10.0, (long) i-1);

        if (i == n) a = x;
        if (i == n-1) b = x;

        if (i % 2 == 0) x = 2*x;

        x = breakdown(x);

        sum += x;


    }

    if (sum % 10 == 0)
    {
        if (a == 4 && (n==13 || n==16))
        {
            printf("VISA\n");
        } else if (a == 3 && (b == 4 || b == 7) && (n==15))
        {
            printf("AMEX\n");
        } else if (a == 5 && (b == 1 || b == 2 || b == 3 || b == 4 || b == 5) && (n==16))
        {
            printf("MASTERCARD\n");
        } else
        {
            printf("INVALID\n");
        }
    } else
    {
        printf("INVALID\n");
    }

}

int length(long number)
{
    int n = 0;
    while (number > 0)
    {
        number = number / 10;
        ++n;
    }

    return n;
}

long power(long base, long exp)
{
    long ans = 1;
    for (int i = 0; i < exp; ++i) {
        ans = ans * base;
    }

    return ans;
}

long breakdown(long x)
{
    if (x / 10.0 >= 1)
    {
        int a = (int) x / 10;
        int b = (int)x % 10;

        return (long) a + (long) b;
    } else
    {
        return x;
    }
}

File: Week 6/PSet/sentimental-credit/credit.py
from cs50 import get_string

number = get_string("Number: ")
n = len(number)

sum = 0
for i in range(n-2,-1,-2):
    to_add = int(number[i])*2
    if len(str(to_add)) >= 2:
        for i in range(len(str(to_add))):
            sum += int(str(to_add)[i])
    else:
        sum += to_add

for i in range(n-1,-1,-2):
    sum += int(number[i])

if str(sum)[-1] == "0":
    if (n == 13 or n == 16) and number[0] == "4":
        print("VISA")
    elif (n == 16) and (number[0] == "5") and (number[1] == "1" or number[1] == "2" or number[1] == "3" or number[1] == "4" or number[1] == "5"):
        print("MASTERCARD")
    elif (n == 15) and (number[0] == "3") and (number[1] == "4" or number[1] == "7"):
        print("AMEX")
    else:
        print("INVALID")
else:
    print("INVALID")
